### PR TITLE
OCPBUGSM-26846 Fix DHCP failing silently on reboot

### PR DESCRIPTION
**Pull request by empovit**
On reboot, NetworkManager doesn't wait for IPv4 addresses if it
gets IPv6 addresses from DHCP. As a result, ignition download
over IPv4 fails. To control this behavior, pass `ip=dhcp` in case
of IPv4, and `ip=dhcp6` in case of IPv6.

### CODE REVIEW

File: internal/host/hostcommands/install_cmd.go [hunk 1]
```diff
@@ -255,35 +256,42 @@ func (i *installCmd) getDiskUnbootableCmd(ctx context.Context, host models.Host)
...
 			return "", err
 		}
-		return string(argsBytes), nil
 	}
 
-	var currentInstallerArgs []string
-	err := json.Unmarshal([]byte(host.InstallerArgs), &currentInstallerArgs)
-	if err != nil {
-		return "", err
+	if cluster.MachineNetworkCidr != "" && !funk.ContainsString(installerArgs, "ip=dhcp6") && !funk.ContainsString(installerArgs, "ip=dhcp") {
```
[the remark this answers is not in the document]
Comment (empovit) [reply internal/host/hostcommands/install_cmd.go:272]: I'll check it out, however
- First, this solution was suggested by the NetworkManager team, and verified in production. Seems like a safe way to go.
- Second, we don't set `--copy-network` when there's no static network configuration (https://github.com/openshift/assisted-service/blob/master/internal/host/hostcommands/install_cmd.go#L261).  On the other hand, we always need either `ip=dhcp` or `ip=dhcp6` to cover DHCP allocation.

[the remark this answers is not in the document]
Comment (empovit) [reply internal/host/hostcommands/install_cmd.go:272]: Also, this setting is needed only on boot, and doesn't affect the installed system https://man7.org/linux/man-pages/man7/dracut.cmdline.7.html. BTW, I'm now wondering how it'll work with a static network configuration.

Comment (ori-amizur) [inline internal/host/hostcommands/install_cmd.go:272]: funk.ContainsString(installerArgs, "ip=dhcp") contains also funk.ContainsString(installerArgs, "ip=dhcp6")

Comment (empovit) [reply internal/host/hostcommands/install_cmd.go:272]: Hmmm, according to https://github.com/thoas/go-funk/blob/master/typesafe.go#L344, it looks for an exact match.

**Comment by empovit** [comment]
@rollandf are there any corner cases you can think of? How is it going to affect none-platform?

File: internal/host/hostcommands/install_cmd_test.go [hunk 2]
```diff
@@ -439,7 +439,91 @@ var _ = Describe("installcmd arguments", func() {
...
+		host    *models.Host
+	)
+	BeforeEach(func() {
+		cluster = &common.Cluster{
+			Cluster: models.Cluster{
+				ImageInfo: &models.ImageInfo{},
+			},
+		}
+		host = &models.Host{}
+	})
```
Comment (flaper87) [inline internal/host/hostcommands/install_cmd_test.go:457]: Could you also add a test for the `StaticNetworkConfig` case? To make sure that `--copy-network` is added and one where `--copy-network` is already present.

**Review by flaper87 — CHANGES_REQUESTED** [review]
2 more  tests just for coverage and sanity

File: internal/host/hostcommands/install_cmd_test.go (hunk 2 above)
Comment (empovit) [reply internal/host/hostcommands/install_cmd_test.go:457]: These have been already covered at a slightly higher level https://github.com/openshift/assisted-service/blob/11e32547a904a2ba49562b4b0ba3a63e09d50d81/internal/host/hostcommands/install_cmd_test.go#L417. Or did you mean something else?

**Review by flaper87 — APPROVED** [review]
:spaghetti:

**Review by flaper87 — APPROVED** [review]

**Pull request merged**